variables/String from array of objects.tpl

```diff
@@ -19,40 +19,55 @@ ___TEMPLATE_PARAMETERS___
 
 [
   {
-    "type": "LABEL",
+    "displayName": "Use this variable to turn an array of objects (e.g. <strong>[{id: '123', name: 'first'},{id: '234', name: 'second}]</strong>) into a string (e.g. <strong>'123,234'</strong>). OPTION create a primitive string (e.g. 1,2,3) from an array [1,2,3]",
     "name": "helpText",
-    "displayName": "Use this variable to turn an array of objects (e.g. <strong>[{id: '123', name: 'first'},{id: '234', name: 'second}]</strong>) into a string (e.g. <strong>'123,234'</strong>)."
+    "type": "LABEL"
   },
   {
-    "type": "SELECT",
-    "name": "inputArray",
-    "displayName": "Input array",
     "macrosInSelect": true,
     "selectItems": [],
-    "simpleValueType": true
+    "displayName": "Input array",
+    "simpleValueType": true,
+    "name": "inputArray",
+    "type": "SELECT"
   },
   {
-    "type": "TEXT",
-    "name": "keyToConcatenate",
-    "displayName": "Object property",
+    "help": "Convert the entire object to a string. (e.g. [{id: '123', name: 'first'},{id: '234', name: 'second}]) becomes \"{id: '123', name: 'first'},{id: '234', name: 'second}\"). Also use for an array of primities (e.g. [1, 2, 3]) becomes \"1,2,3\")",
+    "defaultValue": true,
     "simpleValueType": true,
+    "name": "entireObject",
```

Comment:
The changes you propose here have a couple of problems:

1. The object conversion doesn't happen as you describe in the help text. An array of objects (e.g. [{id: 123},{id: 234}]) isn't serialized into "{id: 123},{id: 234}" but rather "[Object object],[Object object]". If you want to serialize an object, you need `JSON.stringify`.

2. Adding primitive handling to this variable would make it something it is not (it's convert **array of objects to string**, not **any array to string**). I think for handling primitives you should modify the variable you created "String to array", and make it "String to array, and array of primitives to string", or something similar. So the direction of conversion could be a toggle in your variable. Or, we could modify this variable to handle both directions and all conversion types, but it requires some deliberation.

Reply:
Excellent points. I have done as you suggested and added the array of primitives option to my variable.